exporter/otlp-http/opentelemetry-exporter-otlp-http.gemspec

```diff
@@ -29,6 +29,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'opentelemetry-common', '~> 0.19.6'
   spec.add_dependency 'opentelemetry-exporter-otlp-common'
   spec.add_dependency 'opentelemetry-sdk', '~> 1.2'
+  spec.add_dependency 'opentelemetry-semantic_conventions'
```

Comment:
as @ahayworth pointed out to me, i _think_ this needs to be version constrained ala 

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/7cc8a21f80eb77519de0612c718d7a2702692604/instrumentation/trilogy/opentelemetry-instrumentation-trilogy.gemspec#L30

But im not really sure what version? i guess whatever we plan to release the semconv gem as (a patch release? minor? not sure)

Reply:
No problem!, to give you a bit of context I based my commit on the [examples of this repo](https://github.com/open-telemetry/opentelemetry-ruby/search?q=opentelemetry-semantic_conventions+extension%3Agemspec), which are not pinned.

Just tried to stay consistent, that's all :)